diff --git a/src/borg/helpers/fs.py b/src/borg/helpers/fs.py
index e187257ed..67abe52cf 100644
--- a/src/borg/helpers/fs.py
+++ b/src/borg/helpers/fs.py
@@ -79,19 +79,19 @@ def get_keys_dir(*, legacy=False, create=True):
     keys_dir = os.environ.get("BORG_KEYS_DIR")
     if keys_dir is None:
         # note: do not just give this as default to the environment.get(), see issue #5979.
-        keys_dir = str(Path(get_config_dir(legacy=legacy)) / "keys")
+        keys_dir = str(Path(get_config_dir(legacy=legacy, create=create)) / "keys")
     if create:
         ensure_dir(keys_dir)
     return keys_dir
 
 
 def get_security_dir(repository_id=None, *, legacy=False, create=True):
     """Determine where to store local security information."""
     security_dir = os.environ.get("BORG_SECURITY_DIR")
     if security_dir is None:
         get_dir = get_config_dir if legacy else get_data_dir
         # note: do not just give this as default to the environment.get(), see issue #5979.
-        security_dir = str(Path(get_dir(legacy=legacy)) / "security")
+        security_dir = str(Path(get_dir(legacy=legacy, create=create)) / "security")
     if repository_id:
         security_dir = str(Path(security_dir) / repository_id)
     if create:
